ambry-frontend/src/main/java/com/github/ambry/frontend/NamedBlobPutHandler.java

```diff
@@ -196,10 +202,12 @@ private Callback<Void> securityPostProcessRequestCallback(BlobInfo blobInfo) {
               new RetainingAsyncWritableChannel(frontendConfig.maxJsonRequestSizeBytes);
           restRequest.readInto(channel, fetchStitchRequestBodyCallback(channel, blobInfo));
         } else {
+          if (RestUtils.isDatasetUpload(restRequest.getArgs())) {
```

Comment:
we don't have to limit ourself to entire blob upload, we can do the same thing for stitch named blob as well.

Reply:
Sure, but since this pr is sort of huge and considering the priority of the feature, I would prefer to implement and adding corresponding test in future pr.